src/lang/modifyAst/gdt.ts

```diff
@@ -239,6 +241,190 @@ export function addFlatnessGdt({
   }
 }
 
+/**
+ * Adds datum GD&T annotation to the AST.
+ * Creates a single gdt::datum call for a selected face.
+ * Always adds annotation at the end of the AST body.
+ *
+ * @param ast - The AST to modify
+ * @param artifactGraph - The artifact graph for face lookups
+ * @param faces - Selected face to annotate (only first face selection will be used)
+ * @param name - The datum identifier (e.g., 'A', 'B', 'C')
+ * @param framePosition - Position of the feature control frame [x, y] (optional)
+ * @param framePlane - Plane for displaying the frame (XY, XZ, YZ) (optional)
+ * @param fontPointSize - Font point size for annotation text (optional)
+ * @param fontScale - Scale factor for annotation text (optional)
+ * @param nodeToEdit - Path to node to edit (for edit mode)
+ * @returns Modified AST and path to the created node, or an Error
+ */
+export function addDatumGdt({
+  ast,
+  artifactGraph,
+  faces,
+  name,
+  framePosition,
+  framePlane,
+  fontPointSize,
+  fontScale,
+  nodeToEdit,
+}: {
+  ast: Node<Program>
+  artifactGraph: ArtifactGraph
+  faces: Selections
+  name: string
+  framePosition?: KclCommandValue
+  framePlane?: KclCommandValue | string
+  fontPointSize?: KclCommandValue
+  fontScale?: KclCommandValue
+  nodeToEdit?: PathToNode
+}): Error | { modifiedAst: Node<Program>; pathToNode: PathToNode } {
+  // Clone the AST to avoid mutating the original
+  let modifiedAst = structuredClone(ast)
+
+  // Filter to only include face selections
+  const faceSelections = faces.graphSelections.filter((selection) =>
+    isFaceArtifact(selection.artifact)
+  )
```

Comment:
Don't mind the extra step here, just wondering why this isn't a guarantee from the command bar?

Reply:
ok!